web/html/src/manager/virtualization/pools/virtualization-pool-definition-api.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+import * as React from 'react';
+import Network from 'utils/network';
+import * as Messages from 'components/messages';
+
+type Props = {
+  /** Virtual host server ID */
+  hostid: string,
+  /** Name of the pool for which to get the defintion*/
+  poolName: string,
+  /** Children function rendering the content depending on the request result */
+  children: ({definition: Object, messages: React.Node}) => React.Node,
+};
+
+/** Component calling the Uyuni REST API to get the XML definition of a virtual storage pool */
+export function VirtualizationPoolDefinitionApi(props: Props) {
+  const [messages, setMessages] = React.useState([]);
+  const [definition, setDefinition] = React.useState(null);
+
+  React.useEffect(() => {
+    Network.get(`/rhn/manager/api/systems/details/virtualization/pools/${props.hostid}/pool/${props.poolName}`,
+      'application/json').promise
+      .then((response) => {
+        setDefinition(response);
+      }, (xhr) => {
+        const errMessages = xhr.status === 0
```

Comment:
Nitpick: This code seems to repeat across your PRs. Maybe you can extract it to utility function after the PRs with similar code are merged.

Reply:
yes, I'll try to do that after the PRs are merged